dio/lib/src/transformers/fused_transformer.dart

```diff
@@ -111,6 +118,10 @@ class FusedTransformer extends Transformer {
     // of the response or the length of the eagerly decoded response bytes
     final int contentLength;
 
+    // Successful HEAD requests don't have a response body, even if the content-length header
+    // present.
+    final contentLengthHeaderImpliesContent = options.method != 'HEAD';
```

Comment:
- With `CONNECT` too although it's an edge case?
- Also the MDN reference would be helpful to be placed here.
- Could the name be `neverHasResponseBody` and then flip all conditions to the opposite?

```dart
final neverHasResponseBody = options.method == 'CONNECT' || options.method == 'HEAD';

if (neverHasResponseBody || !hasContentLengthHeader) {
  // ...
}
```

Reply:
The issue only occurs when  the `ResponseType` is `.json`, the response body has a Content-Type of `application/json`, and a Content-Length header with a value > 0, but no response body.

A valid `CONNECT` response will have no Content-Type and no Content-Length header, so the transformer won't try to parse it as json.

> A server MUST NOT send any Transfer-Encoding or [Content-Length](https://www.rfc-editor.org/rfc/rfc9110#field.content-length) header fields in a [2xx (Successful)](https://www.rfc-editor.org/rfc/rfc9110#status.2xx) response to CONNECT. A client MUST ignore any Content-Length or Transfer-Encoding header fields received in a successful response to CONNECT.

I mean sure, we could also support invalid responses to CONNECT, but I'd argue if a user want's that, they should just set the `ResponseType` to `.plain, .bytes or .stream.` or use their own decoder / transformer.

Reply:
Cool, thanks for the explanation. But the flag should be flipped because if you are using its opposite result always you should simply flip it.

Reply:
true. done